test/index.test.js

```diff
@@ -0,0 +1,1045 @@
+                assert.notCalled(gitlabScmMock.dummyFunction);
+                assert.notCalled(exampleScmMock.dummyFunction);
+                assert.strictEqual(err, 'there is no suitable webhook module');
+            })
+            .then(null, done())
+            .catch((err) => {
+                done(err);
+            });
+        });
+    });
+
+    describe('chooseScm', () => {
+        const config = { scmContext: 'example.context' };
+
+        it('choose a scm module', () =>
+            scm.chooseScm(config)
+            .then(module => module.dummyFunction(config))
+            .then((result) => {
+                assert.strictEqual(result, 'example');
+                assert.notCalled(githubScmMock.dummyFunction);
+                assert.notCalled(gitlabScmMock.dummyFunction);
+                assert.calledOnce(exampleScmMock.dummyFunction);
+                assert.calledWith(exampleScmMock.dummyFunction, config);
+            }).catch((err) => {
+                assert.fail(err);
+            })
+        );
+
+        it('reject when the scmContext config does not exist', () =>
+            scm.chooseScm({ somekey: 'somevalue' })
+            .then(module => module.dummyFunction())
+            .then(() => {
+                assert.fail();
+            }).catch((err) => {
+                assert.strictEqual(err, 'Not implemented');
+            })
+        );
+
+        it('reject when not registered appropriate scm plugin', () =>
+            scm.chooseScm({ scmContext: 'hoge.cotext' })
+            .then(module => module.dummyFunction())
+            .then(() => {
+                assert.fail();
+            }, (err) => {
+                assert.notCalled(githubScmMock.dummyFunction);
+                assert.notCalled(gitlabScmMock.dummyFunction);
+                assert.notCalled(exampleScmMock.dummyFunction);
+                assert.strictEqual(err, 'Not implemented');
+            }).catch((err) => {
+                assert.fail(err);
+            })
+        );
+    });
+
+    describe('allScm', () => {
+        const config = { scmContext: 'example.context' };
+        const bell = { github: 'githubBell', example: 'exampleBell', gitlab: 'gitlabBell' };
+
+        it('call all origin scm module and return conbined', () =>
+            scm.allScm(module => module.getBellConfiguration(config))
+            .then((result) => {
+                assert.deepEqual(result, bell);
+                assert.calledOnce(githubScmMock.getBellConfiguration);
+                assert.calledWith(githubScmMock.getBellConfiguration, config);
+                assert.calledOnce(exampleScmMock.getBellConfiguration);
+                assert.calledWith(exampleScmMock.getBellConfiguration, config);
+                assert.calledOnce(gitlabScmMock.getBellConfiguration);
+                assert.calledWith(gitlabScmMock.getBellConfiguration, config);
+            }).catch((err) => {
+                assert.throws(new Error(err));
```

Comment:
incorrect usage of `assert.throws`. Also, this `.catch` would "hide" assertion failures in the `.then` above. Instead, just use mocha's built in promise support and don't catch anything here.